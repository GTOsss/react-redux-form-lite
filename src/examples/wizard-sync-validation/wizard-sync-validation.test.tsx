import React from 'react';
import {mount} from 'enzyme';
import {Provider} from 'react-redux';
import {createStore, applyMiddleware, combineReducers} from 'redux';
import ReduxThunkTester from 'redux-thunk-tester';
import WizardSyncValidation from './wizard-sync-validation';
import {reducer} from '../../index';

interface IValuesForm1 {
  firstName?: string;
  lastName?: string;
}

interface IValuesForm2 {
  phone?: string;
  email?: string;
}

interface IValuesForm3 {
  dob?: string;
  hobby?: string;
}

const renderComponent = (onSubmit?) => {
  const reduxThunkTester = new ReduxThunkTester();

  const store = createStore(
    combineReducers({reduxForm: reducer}),
    applyMiddleware(reduxThunkTester.createReduxThunkHistoryMiddleware()),
  );

  const component = mount(
    <Provider store={store}><WizardSyncValidation onSubmit={onSubmit} /></Provider>,
  );

  return {
    reduxThunkTester,
    getStep1: (): IReduxFormState<IValuesForm1> => store.getState().reduxForm.step1 as IReduxFormState<IValuesForm1>,
    getStep2: (): IReduxFormState<IValuesForm2> => store.getState().reduxForm.step2 as IReduxFormState<IValuesForm2>,
    getStep3: (): IReduxFormState<IValuesForm3> => store.getState().reduxForm.step3 as IReduxFormState<IValuesForm3>,
    getWizardExample: (): IReduxFormWizard<IValuesForm1 & IValuesForm2 & IValuesForm3> =>
      store.getState().reduxForm.wizardExample as IReduxFormWizard<IValuesForm1 & IValuesForm2 & IValuesForm3>,
    component,
  };
};

describe('<WizardSyncValidation />', () => {
  test('Render WizardSyncValidation (inputs)', () => {
    const {component} = renderComponent();
    expect(component.find('input')).toMatchSnapshot();
  });

  test('Render WizardSyncValidation: store of wizard', () => {
    const {getWizardExample} = renderComponent();
    expect(getWizardExample()).toMatchSnapshot();
  });

  test('submit failed on page 1 (render inputs)', () => {
    const {component} = renderComponent();
    component.find('form').simulate('submit');
    expect(component.find('input')).toMatchSnapshot();
  });

  test('submit failed on page 1 (store)', () => {
    const {component, getStep1} = renderComponent();
    component.find('form').simulate('submit');
    expect(getStep1()).toMatchSnapshot();
  });

  test('submit failed on page 1 (errors in wizard === errors in step1)', () => {
    const {component, getWizardExample, getStep1} = renderComponent();
    component.find('form').simulate('submit');
    expect(getStep1().form.errorsMap).toStrictEqual(getWizardExample().wizard.errorsMap);
    expect(getWizardExample()).toMatchSnapshot();
  });

  test('submit success on page 1 (render inputs)', () => {
    const {component} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    expect(component.find('input')).toMatchSnapshot();
  });

  test('submit success on page 1 (store of wizard)', () => {
    const {component, getWizardExample} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    expect(getWizardExample()).toMatchSnapshot();
  });

  test('submit success on page 1 (store of step1)', () => {
    const {component, getStep1} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    expect(getStep1()).toMatchSnapshot();
  });

  test('submit success on page 1 (store of step2)', () => {
    const {component, getStep2} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    expect(getStep2()).toMatchSnapshot();
  });

  test('submit success on 1, failed on 2 (render inputs)', () => {
    const {component} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('form').simulate('submit');
    expect(component.find('input')).toMatchSnapshot();
  });

  test('submit success on 1, failed on 2, click prevent page (render inputs)', () => {
    const {component} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('form').simulate('submit');
    component.find('button').at(0).simulate('click');
    expect(component.find('input')).toMatchSnapshot();
  });

  test('submit success on 1, failed on 2, click prevent page, submit success on 1 (render inputs)', () => {
    const {component} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('form').simulate('submit');
    component.find('button').at(0).simulate('click');
    component.find('form').simulate('submit');
    expect(component.find('input')).toMatchSnapshot();
  });

  test('submit success on 1, failed on 2, click prevent page (store of wizard)', () => {
    const {component, getWizardExample} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('form').simulate('submit');
    component.find('button').at(0).simulate('click');
    expect(getWizardExample()).toMatchSnapshot();
  });

  test('submit success on 1, failed on 2, click prevent page, submit success on 1 (store 2)', () => {
    const {component, getStep2} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('form').simulate('submit');
    component.find('button').at(0).simulate('click');
    component.find('form').simulate('submit');
    expect(getStep2()).toMatchSnapshot();
  });

  test('submit success on 1, failed on 2, click prevent (render inputs)', () => {
    const {component} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('button').at(0).simulate('click');
    expect(component.find('input')).toMatchSnapshot();
  });

  test('submit success on 1, failed on 2, click prevent (store of wizard)', () => {
    const {component, getWizardExample} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('button').at(0).simulate('click');
    expect(getWizardExample()).toMatchSnapshot();
  });

  test('submit success on 1, failed on 2 (store 2)', () => {
    const {component, getStep2} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('form').simulate('submit');
    expect(getStep2()).toMatchSnapshot();
  });

  test('submit success on 1, failed on 2 (store of wizard)', () => {
    const {component, getWizardExample} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('form').simulate('submit');
    expect(getWizardExample()).toMatchSnapshot();
  });

  test('submit success on 1, 2 (render inputs)', () => {
    const {component} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    expect(component.find('input')).toMatchSnapshot();
  });

  test('submit success on 1, 2 (store of wizard)', () => {
    const {component, getWizardExample} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    expect(getWizardExample()).toMatchSnapshot();
  });

  test('submit success on 1, failed on 2, click prevent, success on 1 (render inputs)', () => {
    const {component} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('button').at(0).simulate('click');
    component.find('form').simulate('submit');
    expect(component.find('input')).toMatchSnapshot();
  });

  test('submit success on 1, 2, 3 (store of wizard)', () => {
    const {component, getWizardExample} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    expect(getWizardExample()).toMatchSnapshot();
  });

  test('submit success on 1, 2, 3 (store 3)', () => {
    const {component, getStep3} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    expect(getStep3()).toMatchSnapshot();
  });

  test('submit success on page 1, 2, 3 (onSubmit)', () => {
    const onSubmit = jest.fn();
    const {component} = renderComponent(onSubmit);
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    expect(onSubmit.mock.calls).toMatchSnapshot();
  });
});
