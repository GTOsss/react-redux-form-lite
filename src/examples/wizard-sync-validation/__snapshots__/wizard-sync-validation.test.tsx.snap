// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WizardSyncValidation /> Render WizardSyncValidation (inputs) 1`] = `
Array [
  <input
    name="firstName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="First Name"
    type="text"
    value=""
  />,
  <input
    name="lastName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Last Name"
    type="text"
    value=""
  />,
]
`;

exports[`<WizardSyncValidation /> Render WizardSyncValidation: action history 1`] = `
"@@react-redux-form-lite/REGISTER_FORM
 meta: {form: 'step1', wizard: 'wizardExample'},
@@react-redux-form-lite/REGISTER_FIELD
 meta: {
	form: 'step1',
	field: 'firstName',
	wizard: 'wizardExample'
}
 payload: {value: undefined},
@@react-redux-form-lite/REGISTER_FIELD
 meta: {
	form: 'step1',
	field: 'lastName',
	wizard: 'wizardExample'
}
 payload: {value: undefined}"
`;

exports[`<WizardSyncValidation /> Render WizardSyncValidation: inputs 1`] = `
Array [
  <input
    name="firstName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="First Name"
    type="text"
    value=""
  />,
  <input
    name="lastName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Last Name"
    type="text"
    value=""
  />,
]
`;

exports[`<WizardSyncValidation /> Render WizardSyncValidation: store of wizard 1`] = `
Object {
  "values": Object {
    "firstName": undefined,
    "lastName": undefined,
  },
  "wizard": Object {
    "errorsMap": Object {},
    "hasErrors": false,
    "hasWarnings": false,
    "warningsMap": Object {},
  },
}
`;

exports[`<WizardSyncValidation /> submit failed on page 1 (errors in wizard === errors in step1) 1`] = `
Object {
  "values": Object {
    "firstName": undefined,
    "lastName": undefined,
  },
  "wizard": Object {
    "errorsMap": Object {
      "firstName": "Field required.",
      "lastName": "Field required.",
    },
    "hasErrors": true,
    "hasWarnings": false,
    "warningsMap": Object {},
  },
}
`;

exports[`<WizardSyncValidation /> submit failed on page 1 (render inputs) 1`] = `
Array [
  <input
    name="firstName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="First Name"
    type="text"
    value=""
  />,
  <input
    name="lastName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Last Name"
    type="text"
    value=""
  />,
]
`;

exports[`<WizardSyncValidation /> submit failed on page 1 (store) 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": false,
    "errorsMap": Object {
      "firstName": "Field required.",
      "lastName": "Field required.",
    },
    "focused": false,
    "hasErrors": true,
    "hasWarnings": false,
    "submitted": true,
    "warningsMap": Object {},
  },
  "meta": Object {
    "firstName": Object {
      "active": false,
      "blurred": false,
      "changed": false,
      "error": "Field required.",
      "focused": false,
      "warning": "",
    },
    "lastName": Object {
      "active": false,
      "blurred": false,
      "changed": false,
      "error": "Field required.",
      "focused": false,
      "warning": "",
    },
  },
  "values": Object {
    "firstName": undefined,
    "lastName": undefined,
  },
}
`;

exports[`<WizardSyncValidation /> submit success on 1, 2 (render inputs) 1`] = `
Array [
  <input
    name="dob"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Date of birthday"
    type="text"
    value=""
  />,
  <input
    name="hobby"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Hobby"
    type="text"
    value=""
  />,
]
`;

exports[`<WizardSyncValidation /> submit success on 1, 2 (store of wizard) 1`] = `
Object {
  "values": Object {
    "dob": undefined,
    "email": "test",
    "firstName": "test",
    "hobby": undefined,
    "lastName": "test",
    "phone": "test",
  },
  "wizard": Object {
    "errorsMap": Object {},
    "hasErrors": false,
    "hasWarnings": false,
    "warningsMap": Object {},
  },
}
`;

exports[`<WizardSyncValidation /> submit success on 1, 2, 3 (store 3) 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": true,
    "errorsMap": Object {},
    "focused": false,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": true,
    "warningsMap": Object {},
  },
  "meta": Object {
    "dob": Object {
      "active": false,
      "blurred": false,
      "changed": true,
      "error": "",
      "focused": false,
      "warning": "",
    },
    "hobby": Object {
      "active": false,
      "blurred": false,
      "changed": true,
      "error": "",
      "focused": false,
      "warning": "",
    },
  },
  "values": Object {
    "dob": "test",
    "hobby": "test",
  },
}
`;

exports[`<WizardSyncValidation /> submit success on 1, 2, 3 (store of wizard) 1`] = `
Object {
  "values": Object {
    "dob": "test",
    "email": "test",
    "firstName": "test",
    "hobby": "test",
    "lastName": "test",
    "phone": "test",
  },
  "wizard": Object {
    "errorsMap": Object {},
    "hasErrors": false,
    "hasWarnings": false,
    "warningsMap": Object {},
  },
}
`;

exports[`<WizardSyncValidation /> submit success on 1, failed on 2 (render inputs) 1`] = `
Array [
  <input
    name="phone"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Phone"
    type="text"
    value=""
  />,
  <input
    name="email"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Email"
    type="text"
    value=""
  />,
]
`;

exports[`<WizardSyncValidation /> submit success on 1, failed on 2 (store 2) 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": false,
    "errorsMap": Object {
      "email": "Field required.",
      "phone": "Field required.",
    },
    "focused": false,
    "hasErrors": true,
    "hasWarnings": false,
    "submitted": true,
    "warningsMap": Object {},
  },
  "meta": Object {
    "email": Object {
      "active": false,
      "blurred": false,
      "changed": false,
      "error": "Field required.",
      "focused": false,
      "warning": "",
    },
    "phone": Object {
      "active": false,
      "blurred": false,
      "changed": false,
      "error": "Field required.",
      "focused": false,
      "warning": "",
    },
  },
  "values": Object {
    "email": undefined,
    "phone": undefined,
  },
}
`;

exports[`<WizardSyncValidation /> submit success on 1, failed on 2 (store of wizard) 1`] = `
Object {
  "values": Object {
    "email": undefined,
    "firstName": "test",
    "lastName": "test",
    "phone": undefined,
  },
  "wizard": Object {
    "errorsMap": Object {
      "email": "Field required.",
      "phone": "Field required.",
    },
    "hasErrors": true,
    "hasWarnings": false,
    "warningsMap": Object {},
  },
}
`;

exports[`<WizardSyncValidation /> submit success on 1, failed on 2, click prevent (render inputs) 1`] = `
Array [
  <input
    name="firstName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="First Name"
    type="text"
    value="test"
  />,
  <input
    name="lastName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Last Name"
    type="text"
    value="test"
  />,
]
`;

exports[`<WizardSyncValidation /> submit success on 1, failed on 2, click prevent (store of wizard) 1`] = `
Object {
  "values": Object {
    "email": undefined,
    "firstName": "test",
    "lastName": "test",
    "phone": "test",
  },
  "wizard": Object {
    "errorsMap": Object {
      "email": "Field required.",
    },
    "hasErrors": true,
    "hasWarnings": false,
    "warningsMap": Object {},
  },
}
`;

exports[`<WizardSyncValidation /> submit success on 1, failed on 2, click prevent page (render inputs) 1`] = `
Array [
  <input
    name="firstName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="First Name"
    type="text"
    value="test"
  />,
  <input
    name="lastName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Last Name"
    type="text"
    value="test"
  />,
]
`;

exports[`<WizardSyncValidation /> submit success on 1, failed on 2, click prevent page (store of wizard) 1`] = `
Object {
  "values": Object {
    "email": undefined,
    "firstName": "test",
    "lastName": "test",
    "phone": undefined,
  },
  "wizard": Object {
    "errorsMap": Object {
      "email": "Field required.",
      "phone": "Field required.",
    },
    "hasErrors": true,
    "hasWarnings": false,
    "warningsMap": Object {},
  },
}
`;

exports[`<WizardSyncValidation /> submit success on 1, failed on 2, click prevent page, submit success on 1 (render inputs) 1`] = `
Array [
  <input
    name="phone"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Phone"
    type="text"
    value=""
  />,
  <input
    name="email"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Email"
    type="text"
    value=""
  />,
]
`;

exports[`<WizardSyncValidation /> submit success on 1, failed on 2, click prevent page, submit success on 1 (store 2) 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": false,
    "errorsMap": Object {},
    "focused": false,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": false,
    "warningsMap": Object {},
  },
  "meta": Object {
    "email": Object {
      "active": false,
      "blurred": false,
      "changed": false,
      "error": "",
      "focused": false,
      "warning": "",
    },
    "phone": Object {
      "active": false,
      "blurred": false,
      "changed": false,
      "error": "",
      "focused": false,
      "warning": "",
    },
  },
  "values": Object {
    "email": undefined,
    "phone": undefined,
  },
}
`;

exports[`<WizardSyncValidation /> submit success on 1, failed on 2, click prevent, success on 1 (render inputs) 1`] = `
Array [
  <input
    name="phone"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Phone"
    type="text"
    value="test"
  />,
  <input
    name="email"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Email"
    type="text"
    value=""
  />,
]
`;

exports[`<WizardSyncValidation /> submit success on page 1 (action history) 1`] = `
"@@react-redux-form-lite/REGISTER_FORM
 meta: {form: 'step1', wizard: 'wizardExample'},
@@react-redux-form-lite/REGISTER_FIELD
 meta: {
	form: 'step1',
	field: 'firstName',
	wizard: 'wizardExample'
}
 payload: {value: undefined},
@@react-redux-form-lite/REGISTER_FIELD
 meta: {
	form: 'step1',
	field: 'lastName',
	wizard: 'wizardExample'
}
 payload: {value: undefined},
@@react-redux-form-lite/CHANGE
 meta: {
	form: 'step1',
	field: 'firstName',
	wizard: 'wizardExample'
}
 payload: {value: 'test'},
@@react-redux-form-lite/UPDATE_VALIDATE_AND_WARNING_MESSAGES
 meta: {form: 'step1', wizard: 'wizardExample'}
 payload: {
	value: {
		errors: {
			firstName: undefined,
			lastName: 'Field required.'
		},
		warnings: {lastName: undefined}
	},
	submitted: undefined
},
@@react-redux-form-lite/CHANGE
 meta: {
	form: 'step1',
	field: 'lastName',
	wizard: 'wizardExample'
}
 payload: {value: 'test'},
@@react-redux-form-lite/UPDATE_VALIDATE_AND_WARNING_MESSAGES
 meta: {form: 'step1', wizard: 'wizardExample'}
 payload: {
	value: {
		errors: {firstName: undefined, lastName: undefined},
		warnings: {lastName: undefined}
	},
	submitted: undefined
},
@@react-redux-form-lite/UPDATE_FORM_STATE
 meta: {form: 'step1', wizard: 'wizardExample'}
 payload: {
	value: {
		form: {
			submitted: true,
			focused: false,
			blurred: false,
			changed: true,
			errorsMap: {},
			hasErrors: false,
			hasWarnings: false,
			warningsMap: {},
			activeField: ''
		},
		values: {firstName: 'test', lastName: 'test'},
		meta: {
			firstName: {
				focused: false,
				active: false,
				blurred: false,
				changed: true,
				warning: '',
				error: ''
			},
			lastName: {
				focused: false,
				active: false,
				blurred: false,
				changed: true,
				warning: '',
				error: ''
			}
		}
	},
	wizardState: {
		wizard: {
			errorsMap: {},
			hasErrors: false,
			warningsMap: {},
			hasWarnings: false
		},
		values: {firstName: 'test', lastName: 'test'}
	}
},
@@react-redux-form-lite/REGISTER_FORM
 meta: {form: 'step2', wizard: 'wizardExample'},
@@react-redux-form-lite/REGISTER_FIELD
 meta: {
	form: 'step2',
	field: 'phone',
	wizard: 'wizardExample'
}
 payload: {value: undefined},
@@react-redux-form-lite/REGISTER_FIELD
 meta: {
	form: 'step2',
	field: 'email',
	wizard: 'wizardExample'
}
 payload: {value: undefined}"
`;

exports[`<WizardSyncValidation /> submit success on page 1 (render inputs) 1`] = `
Array [
  <input
    name="phone"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Phone"
    type="text"
    value=""
  />,
  <input
    name="email"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Email"
    type="text"
    value=""
  />,
]
`;

exports[`<WizardSyncValidation /> submit success on page 1 (store of step1) 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": true,
    "errorsMap": Object {},
    "focused": false,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": true,
    "warningsMap": Object {},
  },
  "meta": Object {
    "firstName": Object {
      "active": false,
      "blurred": false,
      "changed": true,
      "error": "",
      "focused": false,
      "warning": "",
    },
    "lastName": Object {
      "active": false,
      "blurred": false,
      "changed": true,
      "error": "",
      "focused": false,
      "warning": "",
    },
  },
  "values": Object {
    "firstName": "test",
    "lastName": "test",
  },
}
`;

exports[`<WizardSyncValidation /> submit success on page 1 (store of step2) 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": false,
    "errorsMap": Object {},
    "focused": false,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": false,
    "warningsMap": Object {},
  },
  "meta": Object {
    "email": Object {
      "active": false,
      "blurred": false,
      "changed": false,
      "error": "",
      "focused": false,
      "warning": "",
    },
    "phone": Object {
      "active": false,
      "blurred": false,
      "changed": false,
      "error": "",
      "focused": false,
      "warning": "",
    },
  },
  "values": Object {
    "email": undefined,
    "phone": undefined,
  },
}
`;

exports[`<WizardSyncValidation /> submit success on page 1 (store of wizard) 1`] = `
Object {
  "values": Object {
    "email": undefined,
    "firstName": "test",
    "lastName": "test",
    "phone": undefined,
  },
  "wizard": Object {
    "errorsMap": Object {},
    "hasErrors": false,
    "hasWarnings": false,
    "warningsMap": Object {},
  },
}
`;

exports[`<WizardSyncValidation /> submit success on page 1, 2, 3 (onSubmit) 1`] = `
Array [
  Array [
    Object {
      "actions": Object {
        "arrayPush": [Function],
        "blur": [Function],
        "change": [Function],
        "changeSubmitted": [Function],
        "focus": [Function],
        "registerField": [Function],
        "registerForm": [Function],
        "removeField": [Function],
        "removeForm": [Function],
        "setInitialValues": [Function],
        "updateFormState": [Function],
        "updateValidateAndWarningMessages": [Function],
        "updateValidateMessage": [Function],
        "updateValidateMessages": [Function],
        "updateWarningMessage": [Function],
        "updateWarningMessages": [Function],
      },
      "state": Object {
        "form": Object {
          "activeField": "",
          "blurred": false,
          "changed": true,
          "errorsMap": Object {},
          "focused": false,
          "hasErrors": false,
          "hasWarnings": false,
          "submitted": true,
          "warningsMap": Object {},
        },
        "meta": Object {
          "dob": Object {
            "active": false,
            "blurred": false,
            "changed": true,
            "error": "",
            "focused": false,
            "warning": "",
          },
          "hobby": Object {
            "active": false,
            "blurred": false,
            "changed": true,
            "error": "",
            "focused": false,
            "warning": "",
          },
        },
        "values": Object {
          "dob": "test",
          "hobby": "test",
        },
      },
      "values": Object {
        "dob": "test",
        "hobby": "test",
      },
      "wizard": Object {
        "values": Object {
          "dob": "test",
          "email": "test",
          "firstName": "test",
          "hobby": "test",
          "lastName": "test",
          "phone": "test",
        },
        "wizard": Object {
          "errorsMap": Object {},
          "hasErrors": false,
          "hasWarnings": false,
          "warningsMap": Object {},
        },
      },
    },
  ],
]
`;
